packages/ember-glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -22,4 +26,431 @@ moduleFor('Helpers test: custom helpers', class extends RenderingTest {
     this.assertText('hello world');
   }
 
+  ['@htmlbars helper can recompute a new value']() {
+    let destroyCount = 0;
+    let count = 0;
+    let helper;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        helper = this;
+      },
+      compute() {
+        return ++count;
+      },
+      destroy() {
+        destroyCount++;
+        this._super();
+      }
+    });
+
+    this.render('{{hello-world}}');
+
+    this.assertText('1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@htmlbars helper with arg can recompute a new value']() {
```

Comment:
Nit: I would probably call this helper *with static arguments* can... (and the next test *with dynamic arguments* can ..., to differentiate but also maintaining the parallel)